logic/transactionPool.js

```diff
@@ -276,9 +271,11 @@ TransactionPool.prototype.countUnconfirmed = function () {
  * @param {transaction} transaction
  */
 TransactionPool.prototype.addBundledTransaction = function (transaction) {
-	self.bundled.transactions.push(transaction);
-	var index = self.bundled.transactions.indexOf(transaction);
-	self.bundled.index[transaction.id] = index;
+	if (self.bundled.index[transaction.id] === undefined) {
+		self.bundled.transactions.push(transaction);
+		var index = self.bundled.transactions.indexOf(transaction);
```

Comment:
@MaciejBaj yep, looks like it works as expected, return correct indexes when we pass an object. https://jsfiddle.net/32pwm39f/